draft-ietf-acme-acme.md

```diff
@@ -784,14 +797,19 @@ certificate (optional, string):
     "https://example.com/acme/authz/2345"
   ],
 
+  "finalizeURL": "https://example.com/acme/acct/1/order/1/finalize",
+
   "certificate": "https://example.com/acme/cert/1234"
 }
 ~~~~~~~~~~
 
-The elements of the "authorizations" array are immutable once set.  The server
-MUST NOT change the contents of the "authorizations" array after it is created.
-If a client observes a change in the contents of the "authorizations" array,
-then it SHOULD consider the order invalid.
+Identifiers in a new-order request MAY contain wildcard identifiers, even if
+the resulting authorizations returned by the server do not.
```

Comment:
I think that you need to be a little more precise about defining the wildcard identifier for each type of identifier (i.e., the leftmost label can be '*' for DNS)

Reply:
@martinthomson Thanks for the feedback. I took a shot at clarifying with https://github.com/ietf-wg-acme/acme/pull/342/commits/58eec446e678fa5aec7cf35d44162617f0aa6427 Open to further refinement!

Should there be an RFC reference for the requirements placed on wildcard DNS identifiers? If so what's the best RFC to cite?

I thought that RFC5280 might specify the "only one wildcard character, only as the entire left-most DNS label" behaviour but it [explicitly doesn't address the semantics of wildcard SAN names](https://tools.ietf.org/html/rfc5280#section-4.2.1.6). RFC6125 has a section on [checking wildcard certificates](https://tools.ietf.org/html/rfc6125#section-6.4.3) but it seems a little loosey-goosey and the last MAY is (to my understanding) not something browsers do.

Reply:
I've done some digging on this, and it's a mess. RFC 5280 says dNSName SANs must be in RFC 1034 "preferred name syntax." Preferred name syntax is letter-digit-hyphen, no "*" allowed. Inserting "*"'s became common practice (against 5280), as documented in RFC 6125.

[Section 7.2 of RFC 6125](https://tools.ietf.org/html/rfc6125#section-7.2) describes the situation for issuers, versus 6.4.3, which you link to and which describes the situation for relying parties:
> This document states that the wildcard character '*' SHOULD NOT be
> included in presented identifiers but MAY be checked by application
> clients (mainly for the sake of backward compatibility with deployed
> infrastructure).

The Baseline Requirements say:

> Domain Name: The label assigned to a node in the Domain Name System.
> Fully-Qualified Domain Name: A Domain Name that includes the labels of all superior nodes in the Internet Domain Name System.
> Wildcard Domain Name: A Domain Name consisting of a single asterisk character followed by a single full stop character (“*.”) followed by a Fully-Qualified Domain Name.
> 7.1.4.2.1 Subject Alternative Name Extension
> ...
> Wildcard FQDNs are permitted.

So, AFAICT, RFC5280 doesn't allow wildcards, then RFC6125 extra discourages them, but they are in widespread practical use. I think your current revision is very close. Here's my proposal to bring it closer to the BR definition:

> Any identifier of type "dns" in a new-order request MAY have a wildcard domain
> name as its value. A wildcard domain name consists of a single asterisk
> character followed by a single full stop character (“*.”) followed by a
> domain name as defined for use in the Subject Alternative Name Extension by RFC
> 5280.

Reply:
> I've done some digging on this, and it's a mess. 

Classic.

> So, AFAICT, RFC5280 doesn't allow wildcards, then RFC6125 extra discourages them, but they are in widespread practical use. I think your current revision is very close. Here's my proposal to bring it closer to the BR definition

Thanks! That reads much better. I stole your wording for f451f2c and preserved the "Your wildcard shouldn't be in the returned authorizations" part by more closely matching the MUST NOT from 7.1.4.